### DISCUSSION TITLE
Feature Page - settings not being retained

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#26] [Fixed Issues]
@rern 

I decided to move my Living Room 'production' system to rAudio (RPi3B+ and Merus Amp DAC) so flashed a new card with the latest RPi 2/3 Image (20201224) and  carried out a manual configuration from my Windows 10 PC. All went well until I got to the Features page where SnapClient, Browser on RPi and Filesharing where all enabled by default. I only needed Browser on RPi to use the official 7in Touchscreen, as I had been doing with RA+Re6, so disabled all the other items. On reboot the rAudio logo was displayed on the screen eventually followed by the Arch Linux login prompt but NO web pages was displayed. Back to features page on W10 browser and all the same items were enabled. Reconfigured Features page again and on reboot back to same items enabled and no web page on RPi touchscreen screen. Items on Browser on RPi config subscreen were kept.

Chris

**Comment by ceejayemm** [comment]
@rern 

Applied update 20201228, did not appear to make any difference, options reported still set on reboot and RPi touchscreen still unusable and now the Arch Linux login dialogue has gone too.

This update also seems to have made changes to the MPD screen. In the System page I have the I2S device set to 'Merus Audio Amp' but on the MPD screen while the Audio Output Device is still 'Merus Audio Amp', the Mixer Device is blank and the Mixer Device Volume (under the new small speaker icon) is set to 100 and resets to 100 if it is changed.

Chris

**Reply by ceejayemm** [reply]
@rern 

Further information:  MPD Page / Status reports:

Problem summary:
Failed to open mixer for 'MERUS Audio Amp': no such mixer control:
alsa_output: Decoder is too slow; playing silence to avoid xrun
keepalive request failed for 'http://bbcfmt...1.ts' with error: 'Invalid argument' when opening url, retrying with new connection

Full details:
Dec 28 13:05:52 RAConservatory taskset[2224]: avahi: Service 'Music Player @ RAConservatory' successfully established.
Dec 28 13:13:06 RAConservatory taskset[2224]: ffmpeg/http: Cannot reuse HTTP connection for different host: bbcfmt-ic-52451b00-054d9a-ashlsuklivebbcfmt.s.loris.llnwd.net:-1 != as-hls-uk-live.bbcfmt.s.llnwi.net:-1
Dec 28 13:13:06 RAConservatory taskset[2224]: ffmpeg/hls: keepalive request failed for 'http://as-hls-uk-live.bbcfmt.s.llnwi.net/pool_904/live/uk/bbc_radio_two/bbc_radio_two.isml/bbc_radio_two-audio=320000-251431440.ts' with error: 'Invalid argument' when opening url, retrying with new connection
Dec 28 13:13:06 RAConservatory taskset[2224]: exception: Failed to open mixer for 'MERUS Audio Amp': no such mixer control:
Dec 28 13:13:16 RAConservatory taskset[2224]: ffmpeg/http: Cannot reuse HTTP connection for different host: bbcfmt-ic-52451b00-030714-ashlsuklivebbcfmt.s.loris.llnwd.net:-1 != bbcfmt-ic-52451b00-03e5ea-ashlsuklivebbcfmt.s.loris.llnwd.net:-1
Dec 28 13:13:16 RAConservatory taskset[2224]: ffmpeg/hls: keepalive request failed for 'http://bbcfmt-ic-52451b00-03e5ea-ashlsuklivebbcfmt.s.loris.llnwd.net/pool_904/live/uk/bbc_radio_two/bbc_radio_two.isml/bbc_radio_two-audio=320000-251431441.ts' with error: 'Invalid argument' when opening url, retrying with new connection
Dec 28 13:13:22 RAConservatory taskset[2224]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Dec 28 13:13:22 RAConservatory taskset[2224]: ffmpeg/http: Cannot reuse HTTP connection for different host: bbcfmt-ic-52451b00-03e5ea-ashlsuklivebbcfmt.s.loris.llnwd.net:-1 != as-hls-uk-live.bbcfmt.s.llnwi.net:-1
Dec 28 13:13:22 RAConservatory taskset[2224]: ffmpeg/hls: keepalive request failed for 'http://as-hls-uk-live.bbcfmt.s.llnwi.net/pool_904/live/uk/bbc_radio_two/bbc_radio_two.isml/bbc_radio_two-audio=320000.norewind.m3u8' with error: 'Invalid argument' when parsing playlist
Dec 28 13:13:22 RAConservatory taskset[2224]: player: played "http://as-hls-uk-live.bbcfmt.s.llnwi.net/pool_904/live/uk/bbc_radio_two/bbc_radio_two.isml/bbc_radio_two-audio=320000.norewind.m3u8"

Chris

**Comment by ceejayemm** [comment]
@rern 

Update: RPi Browser not being displayed, Status shows:

Summary: There seems to be library file missing (error while loading shared libraries: libicui18n.so.67: cannot open shared object file: No such file or directory)

Detail:
# systemctl status localbrowser

● localbrowser.service - Local Browser - Chromium
     Loaded: loaded (/etc/systemd/system/localbrowser.service; enabled; vendor preset: disabled)
     Active: failed (Result: exit-code) since Tue 2020-12-29 09:01:28 GMT; 3min 54s ago
    Process: 6447 ExecStart=/usr/bin/sudo /usr/bin/startx (code=exited, status=0/SUCCESS)
    Process: 6562 ExecStop=/usr/bin/killall Xorg (code=exited, status=1/FAILURE)
   Main PID: 6447 (code=exited, status=0/SUCCESS)

Dec 29 09:01:26 RAConservatory sudo[6481]: modprobe: FATAL: Module g2d_23 not found in directory /lib/modules/5.4.83-1-ARCH
Dec 29 09:01:27 RAConservatory sudo[6503]: /usr/lib/chromium/chromium: error while loading shared libraries: libicui18n.so.67: cannot open shared object file: No such file or directory
Dec 29 09:01:27 RAConservatory sudo[6464]: xinit: connection to X server lost
Dec 29 09:01:27 RAConservatory sudo[6464]: 
Dec 29 09:01:27 RAConservatory sudo[6464]: waiting for X server to shut down
Dec 29 09:01:27 RAConservatory sudo[6465]: (II) Server terminated successfully (0). Closing log file.
Dec 29 09:01:28 RAConservatory sudo[6447]: pam_unix(sudo:session): session closed for user root
Dec 29 09:01:28 RAConservatory killall[6562]: Xorg: no process found
Dec 29 09:01:28 RAConservatory systemd[1]: localbrowser.service: Control process exited, code=exited, status=1/FAILURE
Dec 29 09:01:28 RAConservatory systemd[1]: localbrowser.service: Failed with result 'exit-cod

**Reply by rern** [reply]
Please try the UPDATE branch.

**Reply by ceejayemm** [reply]
Update branch fixes problem with RPI Browser  but not Audio Output Mixer problem.

Thanks

Chris

**Comment by ceejayemm** [comment]
@rern 

To bring this thread back to the original problem. I applied update 20201230 this morning and on the Features page the SnapClient and Filesharing options are still being enabled by default. I disabled these again but the next time I came back to this page they were once again enabled.

Chris

**Comment by rern** [comment]
Fixed.

**Reply by ceejayemm** [reply]
In which Update version ?

Chris
